Core/Sources/Preferences/ChatGPTLanguage.swift

```diff
@@ -0,0 +1,209 @@
+        case .icelandic: return "Íslenska"
+        case .italian: return "Italiano"
+        case .swahili: return "Kiswahili"
+        case .haitianCreole: return "Kreyòl Ayisyen"
+        case .kurdish: return "Kurdî"
+        case .latin: return "Latīna"
+        case .latvian: return "Latviešu valoda"
+        case .luxembourgish: return "Lëtzebuergesch"
+        case .lithuanian: return "Lietuvių kalba"
+        case .hungarian: return "Magyar"
+        case .malagasy: return "Malagasy"
+        case .maltese: return "Malti"
+        case .maori: return "Māori"
+        case .dutch: return "Nederlands"
+        case .norwegian: return "Norsk"
+        case .uzbek: return "O'zbek tili"
+        case .polish: return "Polski"
+        case .portuguese: return "Português"
+        case .romanian: return "Română"
+        case .sesotho: return "Sesotho"
+        case .albanian: return "Shqip"
+        case .slovak: return "Slovenčina"
+        case .slovenian: return "Slovenščina"
+        case .finnish: return "Suomi"
+        case .swedish: return "Svenska"
+        case .tagalog: return "Tagalog"
+        case .tatar: return "Tatarça"
+        case .turkish: return "Türkçe"
+        case .vietnamese: return "Việt ngữ"
+        case .yoruba: return "Yorùbá"
+        case .greek: return "Ελληνικά"
+        case .belarusian: return "Беларуская мова"
+        case .bulgarian: return "Български език"
+        case .kyrgyz: return "Кыр"
+        case .kazakh: return "Қазақ тілі"
+        case .macedonian: return "Македонски јазик"
+        case .mongolian: return "Монгол хэл"
+        case .russian: return "Русский"
+        case .serbian: return "Српски језик"
+        case .tajik: return "Тоҷикӣ"
+        case .ukrainian: return "Українська"
+        case .georgian: return "ქართული"
+        case .armenian: return "Հայերեն"
+        case .yiddish: return "ייִדיש"
+        case .hebrew: return "עברית"
+        case .uyghur: return "ئۇيغۇرچە"
+        case .urdu: return "اردو"
+        case .arabic: return "العربية"
+        case .pashto: return "پښتو"
+        case .persian: return "فارسی"
+        case .nepali: return "नेपाली"
+        case .marathi: return "मराठी"
+        case .hindi: return "हिन्दी"
+        case .bengali: return "বাংলা"
+        case .punjabi: return "ਪੰਜਾਬੀ"
+        case .gujarati: return "ગુજરાતી"
+        case .oriya: return "ଓଡ଼ିଆ"
+        case .tamil: return "தமிழ்"
+        case .telugu: return "తెలుగు"
+        case .kannada: return "ಕನ್ನಡ"
+        case .malayalam: return "മലയാളം"
+        case .sinhala: return "සිංහල"
+        case .thai: return "ไทย"
+        case .lao: return "ພາສາລາວ"
+        case .burmese: return "ဗမာစာ"
+        case .khmer: return "ភាសាខ្មែរ"
+        case .korean: return "한국어"
+        case .chinese: return "中文"
+        case .traditionalChinese: return "繁體中文"
+        case .japanese: return "日本語"
+        }
+    }
+}
+
+extension ChatGPTLanguage: CaseIterable {}
```

Comment:
You could modify this list to use `Locale.current.languageCode` to set the default to the current language.

I don't like my code, but you get the idea. I think the built in DB uses the country code as a key.

``` swift
extension ChatGPTLanguage {
    static func currentLanguage() -> ChatGPTLanguage? {
        guard let languageCode = Locale.current.languageCode else { return nil }
        
        switch languageCode {
        case "so":
            return .somali
        case "af":
            return .afrikaans
        case "az":
            return .azerbaijani
        case "id":
            return .indonesian
        case "ms":
            return .malaysianMalay
        case "my":
            return .malay
        case "jv":
            return .javanese
        case "su":
            return .sundanese
        case "bs":
            return .bosnian
        case "ca":
            return .catalan
        case "cs":
            return .czech
        case "ny":
            return .chichewa
        case "cy":
            return .welsh
        case "da":
            return .danish
        case "de":
            return .german
        case "et":
            return .estonian
        case "en":
            if Locale.current.regionCode == "GB" {
                return .englishUK
            } else {
                return .englishUS
            }
        case "es":
            return .spanish
        case "eo":
            return .esperanto
        case "eu":
            return .basque
        case "fr":
            return .french
        case "ga":
            return .irish
        case "gl":
            return .galician
        case "hr":
            return .croatian
        case "xh":
            return .xhosa
        case "zu":
            return .zulu
        case "is":
            return .icelandic
        case "it":
            return .italian
        case "sw":
            return .swahili
        case "ht":
            return .haitianCreole
        case "ku":
            return .kurdish
        case "la":
            return .latin
        case "lv":
            return .latvian
        case "lb":
            return .luxembourgish
        case "lt":
            return .lithuanian
        case "hu":
            return .hungarian
        case "mg":
            return .malagasy
        case "mt":
            return .maltese
        case "mi":
            return .maori
        case "nl":
            return .dutch
        case "no":
            return .norwegian
        case "uz":
            return .uzbek
        case "pl":
            return .polish
        case "pt":
            return .portuguese
        case "ro":
            return .romanian
        case "st":
            return .sesotho
        case "sq":
            return .albanian
        case "sk":
            return .slovak
        case "sl":
            return .slovenian
        case "fi":
            return .finnish
        case "sv":
            return .swedish
        case "tl":
            return .tagalog
        case "tt":
            return .tatar
        case "tr":
            return .turkish
        case "vi":
            return .vietnamese
        case "yo":
            return .yoruba
        case "el":
            return .greek
        case "be":
            return .belarusian
        case "bg":
            return .bul
```

Reply:
`Locale` also offers APIs to get a human readable name for any language code.